python/lib/cloudutils/serviceConfig.py

```diff
@@ -495,6 +495,30 @@ def config(self):
             logging.debug(formatExceptionInfo())
             return False
 
+class hostConfig(serviceCfgBase):
+    def __init__(self, syscfg):
+        super(hostConfig, self).__init__(syscfg)
+        self.serviceName = "Host"
+
+    def config(self):
+        try:
+            cfo = configFileOps("/etc/cloudstack/agent/agent.properties", self)
+            reservedMemory = float(cfo.getEntry("host.reserved.mem.mb").strip() or 1024)
+            totalMemory = float(bash("awk '/MemTotal/ { printf \"%.3f \\n\", $2/1024 }' /proc/meminfo").getStdout().strip())
+            if totalMemory < reservedMemory or totalMemory < 4096 :
+                raise CloudRuntimeException("CloudStack requires a minimum of 4096 MB and more than %d MB memory since %d MB is reserved" %(reservedMemory, reservedMemory))
```

Comment:
@davidjumani this is not necessary, I can run/use KVM hosts with 2GB RAM and lower the host reserved mem property to say 256/512 mb.

Reply:
By default 1 GB is reserved
https://github.com/apache/cloudstack/blob/main/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java#L1038

Should I then update the docs since it says a min of 4GB is required?
http://docs.cloudstack.apache.org/en/latest/installguide/overview/_requirements.html#host-hypervisor-system-requirements

Reply:
The docs are right for recommendation but this code will break usage on people's test env. I would suggest log as a warning. What do others think @nvazquez @sureshanaparti @DaanHoogland...